server/app/repository/ApplicationRepository.java

```diff
@@ -353,7 +353,13 @@ public void updateDraftApplicationProgram(long applicantId, long programId) {
                   queryProfileLocationBuilder.create("updateDraftApplicationProgram"))
               .findOne();
 
-      // Update the existing draft
+      if (existingDraft == null) {
+        throw new RuntimeException(
+            String.format(
+                "Did not find expected draft application for applicantId=%d, programId=%d.",
```

Comment:
Praise: This is a good error message because it includes the applicant ID and program ID, which should save time debugging :)

(No action required)